pkgs/applications/version-management/git-pw/default.nix

```diff
@@ -0,0 +1,43 @@
+{ lib
+, git
+, python3
+, fetchFromGitHub
+}:
+
+python3.pkgs.buildPythonApplication rec {
+  pname = "git-pw";
+  version = "2.5.0";
+
+  PBR_VERSION = version;
+
+  src = fetchFromGitHub {
+    owner = "getpatchwork";
+    repo = "git-pw";
+    rev = version;
+    hash = "sha256-MUothqfdDALKpHzFy5O67vAfbyZOa11YuWRsvpwrSPY=";
+  };
+
+  propagatedBuildInputs = with python3.pkgs; [
+    pyyaml
+    arrow
+    click
+    requests
+    tabulate
+    pbr
+  ];
+
+  nativeCheckInputs = with python3.pkgs; [
+    pytest
+    pytest-cov
+    git
+  ];
+
+  pythonImportsCheck = [ "git_pw" ];
```

Comment:
Does it make sense to do an imports check for an application?  If this is also usable as a library, then it should live in python-modules and use `toPythonApplication` in all-packages as documented in the Nixpkgs manual.

Reply:
In the past other maintainers and I have added it as additional safe-guard also for applications (usually it's covered by the unit tests already). From my point of view, adding it ensure that it's not forgotten if it's moved to `python-modules` for some reason.

Reply:
I went for `testVersion` as there was no conclusion here, alas.